syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -297,3 +267,53 @@ func (j *archiveParser) discoverPkgsFromNestedArchives(parentPkg *pkg.Package) (
 
 	return pkgs, nil
 }
+
+func packageIdentitiesMatch(p pkg.Package, parentPkg *pkg.Package) bool {
+	pkgKey := uniquePkgKey(&p)
+	parentKey := uniquePkgKey(parentPkg)
+
+	// the name/version pair matches...
+	matchesParentPkg := pkgKey == parentKey
```

Comment:
I'd suggest returning early when possible to simplify reading. For example, if `pkgKey == parentKey` is `true`, nothing else in the function body can change the function's return value. It would be helpful to the reader to express that via an early return.

Reply:
Thanks for this change, it helps a lot!